homeassistant/components/systemmonitor/sensor.py

```diff
@@ -52,71 +57,226 @@
 else:
     CPU_ICON = "mdi:cpu-32-bit"
 
-SENSOR_TYPE_NAME = 0
-SENSOR_TYPE_UOM = 1
-SENSOR_TYPE_ICON = 2
-SENSOR_TYPE_DEVICE_CLASS = 3
-SENSOR_TYPE_MANDATORY_ARG = 4
-
 SIGNAL_SYSTEMMONITOR_UPDATE = "systemmonitor_update"
 
-# Schema: [name, unit of measurement, icon, device class, flag if mandatory arg]
-SENSOR_TYPES: dict[str, tuple[str, str | None, str | None, str | None, bool]] = {
-    "disk_free": ("Disk free", DATA_GIBIBYTES, "mdi:harddisk", None, False),
-    "disk_use": ("Disk use", DATA_GIBIBYTES, "mdi:harddisk", None, False),
-    "disk_use_percent": (
-        "Disk use (percent)",
-        PERCENTAGE,
-        "mdi:harddisk",
-        None,
-        False,
+SENSOR_TYPES: Final[tuple[SystemMonitorSensorEntityDescription, ...]] = (
+    SystemMonitorSensorEntityDescription(
+        key="disk_free",
+        name="Disk free",
+        unit_of_measurement=DATA_GIBIBYTES,
+        icon="mdi:harddisk",
+        device_class=None,
+        state_class=STATE_CLASS_MEASUREMENT,
+        mandatory=False,
```

Comment:
No need for adding defaults

```suggestion
        key="disk_free",
        name="Disk free",
        unit_of_measurement=DATA_GIBIBYTES,
        icon="mdi:harddisk",
        state_class=STATE_CLASS_MEASUREMENT,
```